docs/project-model/project-json-reference.md

```diff
@@ -717,6 +547,379 @@ Type: Boolean
 
 TODO
 
+<a name="compilerName"></a>
+### compilerName
+Type: String
+
+The name of the compiler used for this project. `csc` by default. Currently, `csc` (the C# compiler) or `fsc` (the F# compiler) are supported.
+ 
+For example:
+
+    {
+        "compilerName": "fsc"
+    }
+    
+### compile
+Type: Object
+
+An object containing properties for compilation configuration.
+
+<a name="compile-include"></a>
+#### include
+Type: String or String[] with a globbing pattern.
+
+TODO
```

Comment:
The content for most of these TODO can be found in https://github.com/aspnet/Announcements/issues/175#include-structure